db/Seed/seedbuilder.js

```diff
@@ -3,41 +3,108 @@ if (process.env.NODE_ENV !== 'production')
-  .then(() => db.sync({force:true}))
-  .then(() => Promise.map(TherapistList, (therapist) => db.model('therapist').create(therapist)))
-  .then(() => Promise.map(PatientList, (patient) => db.model('patient')
-    .create(Object.assign(patient, {therapist_id: Math.floor((Math.random() * TherapistList.length) + 1)}))))
-  .then(() => Promise.map(ExerciseList,(exercise) => db.model("exercise")
-     .create(Object.assign(exercise,{therapist_id: Math.floor((Math.random() * TherapistList.length) + 1)}))))
-  .then(() => planModel.bulkCreate(PlanList))
-  .then(() => treatmentModel.bulkCreate(TreatmentList))
-  .then(() => workoutModel.bulkCreate(WorkoutList))
-  .then(function () {
-    console.log("Finished Seeding Database");
+  .then(() => db.sync({ force: true }))
+  // Create initial seed data (no addociations yet)
+  .then(() => {
+    return Promise.all(Object.keys(tables).map(table =>
+      db.Promise.map(tables[table], result => db.model(table).create(result))))
+  })
+  .spread((therapists, patients, exercises, plans) => {
+    const updatedTherapists = [];
+    // Assign plans to patients
+    patients.forEach((patient, i) => {
+      patient.setPlans(randomInstances(2, plans));
+    })
+    // Assign patients and random exercises to the therapists
+    therapists.forEach((therapist, i) => {
+      therapist.setPatients(randomInstances(20, patients));
+      updatedTherapists.push(therapist.setExercises(randomInstances(20, exercises)));
+    })
+    return Promise.all(updatedTherapists)
   })
-  .catch(function (err) {
-    console.error('Issue with Seeding', err, err.stack);
+  .then(() => {
+    return Promise.all([
+      Workout.findAll(),
+      Treatment.findAll(),
+      Plan.findAll({
+        include: [{
+          model: Patient,
+          include:[{ model: Therapist, include:[Exercise] }]
+        }]
+      })
+    ])
+  })
+  .spread((workouts, treatments, plans) => {
+    const updatedPlans = [];
+    for (let i=0; i < plans.length; i++) {
+      const randomTreatments = randomInstances(5, treatments);
+      const days = Array(5).fill(0).map((val, i) => i);
+      for (let j=0; j < randomTreatments.length; j++) {
+        // Update workout in reflection of corresponding treatment
+        const workout = randomInstances(1, workouts)[0];
+        workout.update({
+          plan_id: plans[i].id,
+          patient_id: plans[i].patient_id,
+          created_at: moment().subtract(...randomInstances(1, days), 'days'),
+          time_per_exercise: workoutTime(randomTreatments[j].time_per_exercise)
+        })
+        randomTreatments[j].setWorkouts(workout);
+        // Give each treatment a logical ancestor exercise and patient
+        randomTreatments[j].setExercise(...randomInstances(1, plans[i].patient.therapist.exercises));
+        randomTreatments[j].update({ patient_id: plans[i].patient_id });
+
+      }
+      // Assign treatments to plans
+      updatedPlans.push(plans[i].setTreatments(randomTreatments));
```

Comment:
Updated the model to fix this and changed instance methods accordingly (i.e. using `this` instead of directly passing in instance). Can you confirm this is all good before I merge?